torch/csrc/jit/mobile/import.cpp

```diff
@@ -290,7 +316,16 @@ mobile::Module BytecodeDeserializer::deserialize(
   if (reader_->hasRecord("mobile_debug.pkl")) {
     debug_info_bvals = readArchive("mobile_debug", mcu).toTuple()->elements();
   }
-  parseMethods(bvals, debug_info_bvals, *mcu);
+
+  // TODO: skip reading from "constants.pkl" if there is no tuple
+  // with tensor jit index in constants field in "bytecode.pkl", like
+  // ('tensor_jit_index', (4,)).
+  std::vector<IValue> constant_values_from_jit;
+  if (reader_->hasRecord("constants.pkl")) {
```

Comment:
It may increase the memory footprint. Could we load constants.pkl only when needed?

Reply:
It should be doable. The code might be a bit more complicated though, so I am thinking to make a separate change for it. Do you have a preference?

Reply:
It's fine to have two PRs, but would be great to land them together.

Reply:
In the latest commit, I add a boolean to check if bycode model includes tensor jit index. If it does, it will read constant, otherwise skip. Feel free to let me know your thought.

Reply:
Nice!